Set boosterBoxSize for relevant sealedProduct entries

kodawah opened the pull request:
part of #799

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
+                else if "collector" in product["cleanName"].lower():
+                    sealed_product.booster_box_size = 12
+                    if code in booster_box_size_overrides["collector"].keys():
+                        sealed_product.booster_box_size = int(booster_box_size_overrides["collector"][code])
```
`sealed_product.booster_box_size = int(booster_box_size_overrides.get("collector",{}).get(code, 12)`

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
+        with constants.RESOURCE_PATH.joinpath("booster_box_size_overrides.json").open(encoding="utf-8") as f:
+            booster_box_size_overrides = json.load(f)
+
+        code = self.code.upper()
```
There is no `self.code` in this class, you'll need to pass `set_code: str` in to the function

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
             sealed_product.release_date = product["presaleInfo"].get("releasedOn")
+
+            # XXX: this should be wrapped up in a category field
+            if "booster box" in product["cleanName"].lower():
```
This block should be broken out into its own function

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
+            if "booster box" in product["cleanName"].lower():
+                if "set" in product["cleanName"].lower():
+                    sealed_product.booster_box_size = 30
+                    if code in booster_box_size_overrides["set"].keys():
```
`.keys()` isn't necessary anymore, as it's default behavior (Mypy should've caught this)

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
+
+            # XXX: this should be wrapped up in a category field
+            if "booster box" in product["cleanName"].lower():
+                if "set" in product["cleanName"].lower():
```
`if "Set" in product["cleanName'}` removes the need to lower (same elsewhere)

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +276,33 @@ def generate_mtgjson_sealed_product_objects(
         mtgjson_sealed_products = []
 
+        # Load cache if not loaded
+        with constants.RESOURCE_PATH.joinpath("booster_box_size_overrides.json").open(encoding="utf-8") as f:
```
File not found :O

kodawah committed:
Add a way to fix up sealed product names

kodawah committed:
Set category and subtype fields for each product available

This employs some heuristics and simple string parsing, as well as
arbitrary categorization.

kodawah committed:
Set prductSize field to track unit size of product entries

Currently only supported for booster_box category.

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -256,8 +269,178 @@ def generate_today_price_dict(
+        """
+
+        # The order of the checks is important
+        for tag in [
```
Could simplify this with `if any(tag for tag in list): return CASE`

kodawah replied:
ok

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +459,53 @@ def generate_mtgjson_sealed_product_objects(
+        ) as f:
+            sealed_name_fixup = json.load(f)
+
+        with constants.RESOURCE_PATH.joinpath("booster_box_size_overrides.json").open(
```
this is duplicated of 462

kodawah replied:
fixed

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +459,53 @@ def generate_mtgjson_sealed_product_objects(
+        with constants.RESOURCE_PATH.joinpath("booster_box_size_overrides.json").open(
+            encoding="utf-8"
+        ) as f:
+            booster_box_size_overrides = json.load(f)
```
might not be a good idea to call it booster box, since this is every type of product now?

kodawah replied:
ok!

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +459,53 @@ def generate_mtgjson_sealed_product_objects(
+                f"{sealed_product.name}: {sealed_product.category.value}.{sealed_product.subtype.value}"
+            )
+
+            if sealed_product.category == MtgjsonSealedProductCategory.BOOSTER_BOX:
```
Is this override really needed here?

kodawah replied:
yes sometimes "draft" is not present so we need to fallback on what we know

kodawah replied:
i'll add a comment

kodawah replied:
oh that's not an override, it's just that the data we have only covers the booster box size

ZeldaZach commented on mtgjson5/classes/mtgjson_sealed_product.py:
```diff
@@ -1,13 +1,117 @@
 from .mtgjson_purchase_urls import MtgjsonPurchaseUrlsObject
 
 
+class MtgjsonSealedProductCategory(enum.Enum):
```
I'd prefer each of these enums to be defined in their own file, and then use the `__init__` to bring them together

kodawah replied:
i think it makes sense to keep everything related to sealed in a single place no?

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -100,6 +106,13 @@ class TCGPlayerProvider(AbstractProvider):
         "3x Magic Booster Packs",
         "Booster Battle Pack",
     ]
+    booster_box_default_size = {
```
not sure if booster_box is the correct term anymore

kodawah replied:
it's not, moving to product_default_size, but for now not converting it to a dict

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -256,8 +269,194 @@ def generate_today_price_dict(
+        # The order of the checks is important
+        if any(
+            tag in product_name
+            for tag in [
```
Would it make sense to const-ify these lists across the board? Would simplify the code here and maybe let us lambda function this

kodawah replied:
i don't think so? these are just random names that happened to be on tcg and could change at any moment

what did you have in mind?

ZeldaZach replied:
Just a general idea I was thinking

```python3
def helper_with_better_name(find_list, return_value_if_found):
    return value_if_found if any(x for x in find_list) else None
```

and just repeatedly call to that setup. Could also be a lambda
```python3
lambda x, y: return y if any(a for a in x) else None
```

kodawah replied:
i might need an example for this ^^

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -256,8 +269,194 @@ def generate_today_price_dict(
+        :return: category
+        """
+
+        # The order of the checks is important
```
I feel this comment is underrated and needs more context to it on why the order matters

kodawah replied:
ill expand on it

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +475,48 @@ def generate_mtgjson_sealed_product_objects(
+                        sealed_product.subtype.value, {}
+                    ).get(
+                        set_code,
+                        self.product_default_size.get(sealed_product.subtype.value, {}),
```
Getting a default dict if the sealed_product isn't found doesn't seem right. Should be an integer

kodawah replied:
good catch

ZeldaZach commented on mtgjson5/resources/sealed_name_fixes.json:
```diff
@@ -0,0 +1,12 @@
+    "DotP": "Duels of the Planeswalkers",
+    "M15 Prerelease": "Magic 2015 Prerelease",
+    "All 5": "Set of 5",
+    "M10 ": "",
```
Is it intentional the M1X have a space after the name

kodawah replied:
yes, it's needed to remove an extra space

kodawah replied:
if you want i can just remove the space in the tag and then add an extra entry to replace `"  "` with `" "`

ZeldaZach commented on mtgjson5/providers/tcgplayer.py:
```diff
@@ -276,11 +478,48 @@ def generate_mtgjson_sealed_product_objects(
+        ) as f:
+            sealed_name_fixes = json.load(f)
+
+        with constants.RESOURCE_PATH.joinpath("sealed_name_fixes.json").open(
```
This is loading the wrong file
```suggestion
        with constants.RESOURCE_PATH.joinpath("booser_box_size_overrides.json").open(
```

ZeldaZach committed:
Update mtgjson5/providers/tcgplayer.py
